### DISCUSSION TITLE
Add carbon-copy and blind carbon-copy to notification emails

### DISCUSSION THREAD

**Discussion by benjibee** [duncanmcclean/simple-commerce#814] [Feature Requests]
I'm attempting to add CC or BCC email addresses to email notifications sent out using the following code in `config/simple-commerce.php`

```php
\DoubleThreeDigital\SimpleCommerce\Notifications\BackOfficeOrderPaid::class => [
    'to' => env('MAIL_FROM_ADDRESS'),
    'bcc' => 'test@example.com',
]
```

Checking mail messages sent out in Telescope (and my inbox) it doesn't look like the additional addresses are being sent an email. I had a look over the `BackOfficeOrderPaid` class and the [Laravel notification documentation](https://laravel.com/docs/9.x/notifications) but I don't see any reason these emails aren't being used.

**Comment by duncanmcclean** [comment]
Simple Commerce currently only supports the `to` option for configuring notifications, not anything else, see [the code](https://github.com/duncanmcclean/simple-commerce/blob/main/src/Listeners/SendConfiguredNotifications.php#L57).

For `cc` and `bcc` to work there, we'll need to add support for it (so it gets passed along to the notification). It should be fairly easy to do though - I'll see if I can add it tonight if I have some free time.

**Reply by benjibee** [reply]
I looked, I promise! A class with a name like `SendConfiguredNotifications` should have been pretty obvious, though.

If you have a fix and think this might be useful to others, I'd be very happy to take advantage of it. Thank you 😄

**Comment by duncanmcclean** [comment]
I've just had a look into this & unfortunately it doesn't look like something we can implement in the `SendConfiguredNotifications` listener, we can only set the recipient that way.

In order to CC/BCC someone into an email, you'll need to override the `toMail` method on the notification itself. The easiest way to do this is:

1. Create a new file, `app/Notifications/BackOfficeOrderPaid.php`
2. Inside it, copy the contents of the [`BackOfficeOrderPaid`](https://github.com/duncanmcclean/simple-commerce/blob/main/src/Notifications/BackOfficeOrderPaid.php) notification that comes with Simple Commerce. You will need to change the namespace at the top to be `App\Notifications`.
3. Next, inside the `toMail` method, tag on the CC....

```php
public function toMail($notifiable)
{
        return (new MailMessage)
            ->subject(__(':siteName: Order Confirmation', ['siteName' => config('app.name')]))
            ->markdown('simple-commerce::emails.customer_order_paid', [
                'order' => $this->order,
                'site' => $this->order->site(),
            ])
            ->cc('test@example.com');
}
```

4. Finally, replace the reference of the notification in your Simple Commerce config file with your custom class, `App\Notifications\BackOfficeOrderPaid`.

Hopefully that'll work for you!

**Reply by benjibee** [reply]
This works, thanks. I already had a custom class for this notification, anyway. Thanks for the quick replies!
